akamai/edgegrid/edgegrid.py

```diff
@@ -133,27 +131,73 @@ def from_edgerc(rcinput, section='default'):
             max_body=rc.getint(section, 'max_body')
         )
 
+    def handle_redirect(self, res, **kwargs):
+        if res.is_redirect:
+            redirect_location = res.headers['location']
+
+            logger.debug("signing the redirected url: %s", redirect_location)
+            request_to_sign = res.request.copy()
+            request_to_sign.url = redirect_location
+
+            res.request.headers['Authorization'] = self.ah.make_auth_header(
+                request_to_sign.url,
+                request_to_sign.headers,
+                request_to_sign.method,
+                request_to_sign.body,
+                eg_timestamp(),
+                new_nonce()
+            )
+
+    def __call__(self, r):
+        timestamp = eg_timestamp()
+        nonce = new_nonce()
+
+        r.headers['Authorization'] = self.ah.make_auth_header(
+            r.url,
+            r.headers,
+            r.method,
+            r.body,
+            timestamp,
+            nonce
+        )
+        r.register_hook('response', self.handle_redirect)
+        return r
+
+
+class EdgeGridAuthHeaders():
+
+    def __init__(self, client_token, client_secret, access_token,
+                 headers_to_sign=None, max_body=131072):
+        self.client_token = client_token
+        self.client_secret = client_secret
+        self.access_token = access_token
+        if headers_to_sign:
```

Comment:
Instead of having this `if` block here, what do you think of assigning a default value of `headers_to_sign=()` in this method signature, as mentioned in my other method? I have not tested that myself, but probably you may have to apply the same default value in line 88 (EdgeGridAuth constructor).

Thanks :-)